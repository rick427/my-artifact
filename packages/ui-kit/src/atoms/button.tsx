// packages/ui-kit/src/atoms/Button.tsx
import React from 'react';

export interface ButtonProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
  children: React.ReactNode;
}

export const Button = ({ children, ...props }:ButtonProps) => (
  <button style={{ padding: '10px 20px', cursor: "pointer", borderRadius: 6, background: 'red', color: '#fff' }} {...props}>
    {children}
  </button>
);
