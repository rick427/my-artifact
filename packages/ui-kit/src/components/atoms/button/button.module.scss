// button.module.scss

.button {
  padding: 0.75rem 1.5rem;
  border-radius: var(--borderRadius);
  font-family: var(--font-family);
  font-size: var(--font-size);
  color: var(--colors-light);
  background-color: var(--colors-primary);
  cursor: pointer;
  transition: background-color 0.2s ease-in-out;

  &.--outline{
    background-color: transparent;
    border: 1px solid var(--colors-primary);
    color: var(--colors-primary);
  }
}
