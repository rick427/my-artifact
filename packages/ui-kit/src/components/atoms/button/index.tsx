//packages/ui-kit/button

import React from 'react';

import styles from './button.module.scss';

export interface ButtonProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
  children: React.ReactNode;
  variant?: 'primary' | 'outline';
  className?: string;
}

export const Button = (props:ButtonProps) => {
    const { children, variant = 'primary', className = '', ...rest} = props;

    const combinedClassName = [
        styles.button,
        styles[`button--${variant}`],
        className
    ].filter(Boolean).join(' ');

    return (
        <button className={combinedClassName} {...rest}>
            {children}
        </button>
    );
};
