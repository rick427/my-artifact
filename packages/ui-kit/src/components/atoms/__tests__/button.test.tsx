/// <reference types="vitest/globals" />

import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import { Button } from '../button';

describe('Button component', () => {
  it('renders with children', () => {
    render(<Button>Click me</Button>);
    expect(screen.getByText('Click me')).toBeInTheDocument();
  });

  it('fires onClick when clicked', () => {
    const handleClick = vi.fn();
    render(<Button onClick={handleClick}>Click me</Button>);
    fireEvent.click(screen.getByText('Click me'));
    expect(handleClick).toHaveBeenCalledTimes(1);
  });

  it('applies the default primary variant class', () => {
    render(<Button>Primary</Button>);
    const button = screen.getByRole('button', { name: 'Primary' });
    expect(button.className).toMatch(/button--primary/);
  });

  it('applies the outline variant class', () => {
    render(<Button variant="outline">Outline</Button>);
    const button = screen.getByRole('button', { name: 'Outline' });
    expect(button.className).toMatch(/button--outline/);
  });

  it('applies custom className', () => {
    render(<Button className="custom-class">Custom</Button>);
    const button = screen.getByRole('button', { name: 'Custom' });
    expect(button.className).toMatch(/custom-class/);
  });

  it('passes HTML button attributes (e.g. disabled)', () => {
    render(<Button disabled>Disabled</Button>);
    const button = screen.getByRole('button', { name: 'Disabled' });
    expect(button).toBeDisabled();
  });

  it('supports setting button type', () => {
    render(<Button type="submit">Submit</Button>);
    const button = screen.getByRole('button', { name: 'Submit' });
    expect(button).toHaveAttribute('type', 'submit');
  });

  it('supports aria-label and other HTML props', () => {
    render(<Button aria-label="icon button" title="info">Icon</Button>);
    const button = screen.getByRole('button', { name: 'icon button' });
    expect(button).toHaveAttribute('aria-label', 'icon button');
    expect(button).toHaveAttribute('title', 'info');
  });

  it('matches snapshot', () => {
    const { container } = render(<Button variant="primary" className="test-class">Snapshot</Button>);
    expect(container.firstChild).toMatchSnapshot();
  });
});
