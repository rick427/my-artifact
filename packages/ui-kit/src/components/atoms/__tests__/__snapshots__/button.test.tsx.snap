// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button component > matches snapshot 1`] = `
<button
  class="_button_afc71e _button--primary_afc71e test-class"
>
  Snapshot
</button>
`;
